example/hvstack.hpp

```diff
@@ -0,0 +1,95 @@
+#include "boost/gil/image_view_factory.hpp"
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <stdexcept>
+#include <vector>
+
+namespace boost { namespace gil {
+template <typename View>
+void hstack(const std::vector<View>& views, const View& output_view)
+{
+    if (views.size() == 0)
+    {
+        throw std::invalid_argument("empty views vector is passed - cannot create stacked image");
+    }
+
+    auto dimensions = views.front().dimensions();
+    for (const auto& view : views)
+    {
+        if (view.dimensions() != dimensions)
```

Comment:
Maybe we only need to check one dimension. That is, the heights should be same for h-stacking. One advantage could be making a panaroma view.

Reply:
I implemented this feature.